prompting/datasets/wiki.py

```diff
@@ -1,90 +1,64 @@
-# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
-# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
-# DEALINGS IN THE SOFTWARE.
-
 import re
 import sys
 import random
-import datetime
 import bittensor as bt
-import wikipedia as wiki
-from typing import Dict, Union, List, Tuple
+import wikipedia
 from queue import Queue, Full, Empty
 from functools import lru_cache
-from .base import Dataset
-from ..selector import Selector
-
+from prompting.datasets.base import BaseDataset
+from prompting.datasets.base import Context
+from typing import ClassVar
+from typing import Optional
+from pydantic import model_validator, ConfigDict
 
 # Create a queue called CACHED_ARTICLES to store wikipedia articles that have been fetched
-CACHED_ARTICLES = Queue(maxsize=300)
+CACHED_ARTICLES: Queue[Context] = Queue(maxsize=300)
 
 
 # speed up page loading
 @lru_cache(maxsize=1000)
-def _get_page(
-    title, pageid=None, auto_suggest=False, redirect=True, seed=None
-) -> wiki.WikipediaPage:
+def _get_page(title, pageid=None, auto_suggest=False, redirect=True, seed=None) -> wikipedia.WikipediaPage:
     """Cached Wikipedia page loading."""
     try:
-        page = wiki.page(
-            title=title, pageid=pageid, auto_suggest=auto_suggest, redirect=redirect
-        )
+        page = wikipedia.page(title=title, pageid=pageid, auto_suggest=auto_suggest, redirect=redirect)
         # create sections manually if not found
         if not page.sections:
             page._sections = [
-                line.strip("= ")
-                for line in page.content.splitlines()
-                if re.search(r"=+\s+.*\s+=+", line)
+                line.strip("= ") for line in page.content.splitlines() if re.search(r"=+\s+.*\s+=+", line)
             ]
         return page
 
-    except wiki.DisambiguationError as e:
+    except wikipedia.DisambiguationError as e:
         bt.logging.debug(f"{e.__class__.__name__} loading page {title!r}: {e}")
-        # exc info contains a tuple of (requested_title: str, possible_matches: List[str])
+        # exc info contains a tuple of (requested_title: str, possible_matches: list[str])
         pages = sys.exc_info()[1].args[1]
-        if not type(pages) == list:
+        if not isinstance(pages, list):
             return None
         title = random.Random(seed).choice(pages)
         return _get_page(title, auto_suggest=auto_suggest, redirect=redirect)
 
-    except wiki.PageError as e:
+    except wikipedia.PageError as e:
         bt.logging.warning(f"{e.__class__.__name__} loading page {title!r}: {e}")
         if not auto_suggest:
             return _get_page(title, auto_suggest=True, redirect=redirect)
         return None
 
 
 @lru_cache(maxsize=1000)
-def _get_random_titles(pages=10, seed=42) -> List:
+def _get_random_titles(pages=10, seed=42) -> list:
```

Comment:
The previous typing was actually partially correct, but missing the datatype of the list content.  It should have been something like `List[str]` or `List[wikipedia.WikipediaPage]`.  We should be using the `typing` library when possible.

Reply:
as per discussion typing is deprecated since py3.9